### PR TITLE
Add TUF model package

### PR DESCRIPTION
**Pull request by patflynn**
Uses GSON to import TUF metadata.

WIP towards #60.

### CODE REVIEW

File: src/main/java/dev/sigstore/json/GsonSupplier.java [hunk 1]
```diff
@@ -30,13 +30,23 @@
...
-          .registerTypeAdapter(byte[].class, new GsonByteArrayAdapter())
-          .registerTypeAdapterFactory(new GsonAdaptersRekorEntry())
-          .registerTypeAdapterFactory(new GsonAdaptersRekorEntryBody())
-          .disableHtmlEscaping()
-          .create();
+  private final Gson gson;
+
+  GsonSupplier() {
+    GsonBuilder gsonBuilder = new GsonBuilder();
+    for (TypeAdapterFactory factory : ServiceLoader.load(TypeAdapterFactory.class)) {
```
Comment (loosebazooka) [inline src/main/java/dev/sigstore/json/GsonSupplier.java:37]: does this just autoload all the typeadapters in the classpath?

Comment (patflynn) [reply src/main/java/dev/sigstore/json/GsonSupplier.java:37]: yeah pretty much. Immutables registers them as META-INF/services/com.google.gson.TypeAdapterFactory, and then this code just loads them all up.

Comment (patflynn) [reply src/main/java/dev/sigstore/json/GsonSupplier.java:37]: https://immutables.github.io/json.html

Comment (patflynn) [reply src/main/java/dev/sigstore/json/GsonSupplier.java]: I've changed the code back to using explicit registration.

**Review by loosebazooka — APPROVED** [review]
I can't really tell if any types can be "empty" in which case I think they should be Optional<...>. But anyway, lgtm.

**Pull request merged**